src/ibex_bluesky_core/callbacks/plotting.py

```diff
@@ -74,3 +86,49 @@ def start(self, doc: RunStart) -> None:
         """Process an start document (delegate to superclass, then show the plot)."""
         super().start(doc)
         show_plot()
+
+
+class PlotPNGSaver(QtAwareCallback):
+    """Save plots to PNG files on a run end."""
+
+    def __init__(
+        self,
+        x: str,
+        y: str,
+        ax: Axes | None,
+        postfix: str,
+        output_dir: str | os.PathLike[str] | None,
+    ) -> None:
+        """Initialise the PlotPNGSaver callback.
+
+        Args:
+            x: The name of the signal for x.
+            y: The name of the signal for y.
+            ax: The subplot to save to a file.
+            postfix: The file postfix.
+            output_dir: The output directory for PNGs.
+
+        """
+        super().__init__()
+        if ax is None:
+            raise ValueError("ax is None")
```

Comment:
Why do we type-hint `ax: Axes | None,` if we just throw an error if it is `None`?